pkgs/development/compilers/javacard-devkit/default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv, requireFile, unzip, makeWrapper, oraclejdk8 }:
+
+let
+  pkg = "javacard-devkit";
+  version = "2.2.2";
+
+  underscoreVersion = builtins.replaceStrings ["."] ["_"] version;
+
+  downloadUrl = "http://www.oracle.com/technetwork/java/javasebusiness/downloads/java-archive-downloads-javame-419430.html#java_card_kit-2.2.2-oth-JPR";
+in
+
+stdenv.mkDerivation rec {
+  name = "${pkg}-${version}";
+
+  src = requireFile {
+    name = "java_card_kit-${underscoreVersion}-linux.zip";
+    url = downloadUrl;
+    sha256 = "1rzkw8izqq73ifvyp937wnjjc40a40drc4zsm0l1s6jyv3d7agb2";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackPhase = "true";
```

Comment:
Something like this with `oraclejdk8` in `nativeBuildInputs` should unpack the nested Zip file:

```nix
{
  # ...
  sourceRoot = ".";
  zipPrefix = "java_card_kit-${underscoreVersion}";
  unpackCmd = ''
    unzip -p "$curSrc" "$zipPrefix/$zipPrefix-rr-bin-linux-do.zip" | jar x
  '';
  # ...
}
```

Setting `sourceRoot` to `.` prevents unpackPhase from bailing out due to multiple directories.

Reply:
Thanks, didn't know about the `sourceRoot = "."` trick!